source/reflect.h2

```diff
@@ -3722,8 +3911,12 @@ apply_metafunctions: (
         else if name == "regex" {
             regex_gen( rtype );
         }
+        else if name == "_internal_foreign_interface_pseudovalue" {
+            foreign_interface_pseudovalue( rtype );
+        }
+        else if libraries_try_apply(n.fully_qualified_name(), name, rtype) {}
         else {
-            error( "unrecognized metafunction name: " + name );
+            error( "unrecognized metafunction name: '" + name + "' for type declaration" );
```

Comment:
I think the detection logic should be changed. We should have two lists:
 - Internal metafunctions
 - External metafunctions

First look up the internal ones, afterwards the external ones. If both are not found create an error that lists all. 

This would reduce the number of places which need to change when a new metafunction is implemented.

Reply:
I am not sure if we should dump every single metafunction that could be registered, for big programs the list could grow quite big. This does look first on internals and then externals second though.

Reply:
Currently we print out a list of all metafunctions. Printing out all external metafunctions would be a logical consequence. It also helps the user to debug. If he does not find the metafunction in the list it is not loaded. He might had a typo and can see how the metafunction is actually called. 

 - Printing out everything could be enabled via a compiler flag
 -  Or a flag can force cppfront to print out all registered metafunctions.